velox/functions/prestosql/tests/ArraysOverlapTest.cpp

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <optional>
+#include "velox/functions/prestosql/tests/FunctionBaseTest.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::functions::test;
+
+namespace {
+class ArraysOverlapTest : public FunctionBaseTest {
+ protected:
+  void testExpr(
+      const VectorPtr& expected,
+      const std::string& expression,
+      const std::vector<VectorPtr>& input) {
+    auto result =
+        evaluate<SimpleVector<bool>>(expression, makeRowVector(input));
+    assertEqualVectors(expected, result);
+  }
+  template <typename T>
+  void testInt() {
+    auto array1 = makeNullableArrayVector<T>({
+        {1, -2, 3, std::nullopt, 4, 5, 6, std::nullopt},
+        {1, 2, -2, 1},
+        {3, 8, std::nullopt},
+        {1, 1, -2, -2, -2, 4, 8},
+    });
+    auto array2 = makeNullableArrayVector<T>({
```

Comment:
Can you also test arrays with null for "array2" and null arrays for both "array1" and "array2"?